Leanplum-SDK/Classes/Managers/Networking/LPFileTransferManager.m

```diff
@@ -60,6 +67,27 @@ + (instancetype)sharedInstance {
     return sharedManager;
 }
 
+-(LPFeatureFlagManager *)featureFlagManager {
```

Comment:
This isnt thread safe. I am trying to understand how we are using dependency injection. Also, may be dispatch_once might be useful

Reply:
When we instantiate this class in tests, we can set a mock featureFlagManager to this property and that allows us to control the behavior of this dependency. Does that make sense?

Reply:
we can put a dispatch_once here for thread safety. Are you ok with that ?

Reply:
If you look deeper at the code, you will notice that it calls sharedInstance. That method itself is thread-safe, making this method idempotent and therefore thread-safe.